waiter/integration/waiter/deployment_errors_test.clj

```diff
@@ -23,10 +33,18 @@
                    :x-waiter-queue-timeout 600000}
           {:keys [headers body] :as response} (make-request-with-debug-info headers #(make-kitchen-request waiter-url %))
           service-id (get headers "x-waiter-service-id")]
-      (is (not (nil? service-id)))
-      (assert-response-status response 503)
-      (is (str/includes? body (str "Deployment error: " (-> (waiter-settings waiter-url) :messages :invalid-health-check-response))))
-      (delete-service waiter-url service-id))))
+      (try
```

Comment:
We should have an `assert-deployment-error` macro for this.

Reply:
I couldn't get this to work:
<pre>
(defmacro assert-deployment-error
  [waiter-url {:keys [headers body] :as response} message-key]
  `(let [service-id# (get ~headers "x-waiter-service-id")    
         service-state# (get-router->service-state ~waiter-url service-id#)]
     (is service-id#)
     (is service-state#)
     (assert-response-status ~response 503)
     (is (str/includes? ~body (str "Deployment error: " (-> (waiter-settings ~waiter-url)
                                                            :messages
                                                            ~message-key)))
         (str "router->service-state:\n" (with-out-str (pprint/pprint service-state#))))))
</pre>

For some reason the `headers` and `body` were nil.  It worked fine if I avoided the destructuring and passed them as separate params.  If you can figure out, I'm happy to change it.  I'm not keen on factoring tests too much in the first place.

Reply:
I think you can destructure in a let-block and use the `destructure` function. https://clojure.org/guides/destructuring#_macros